src/sim.py

```diff
@@ -161,17 +184,33 @@ def main():
   parser.add_argument('osm_data',           type=str, help='TODO')
   parser.add_argument('depots_filename',    type=str, help='TODO')
   parser.add_argument('output_filename',    type=str, help='TODO')
+  parser.add_argument('--sim-parameters',   type=str, help='TODO') # a yaml config file
   args = parser.parse_args()
 
   print(f"parsed_gtfs_prefix: {args.parsed_gtfs_prefix}")
   print(f"osm_data:           {args.osm_data}")
   print(f"depots_filename:    {args.depots_filename}")
   print(f"output_filename:    {args.output_filename}")
-
-
-  bus_assignments, bus_counts = simulate(args.parsed_gtfs_prefix, args.osm_data, args.depots_filename, args.output_filename)
-
-
+  print(f"sim_parameters:     {args.sim_parameters}")
+
+  # sim_parameters assumed to be a yaml config file of k:v pairs mapping to the 
+  # potential inputs to generateParams()
+  #TODO write results to output_filename
+  if args.sim_parameters is not None:
+    with open(args.sim_parameters,'r') as f:
+      params = yaml.full_load(f)
+      bus_assignments, bus_counts = simulate(args.parsed_gtfs_prefix, 
+                                              args.osm_data,
+                                              args.depots_filename,
+                                              args.output_filename,
+                                              parameters=params
+                                            )
+  else:
```

Comment:
Remove `if/else`. Set `params=None` and then load yaml into it if necessary pass to `simulate`.